File: 1544-count-good-nodes-in-binary-tree/1544-count-good-nodes-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function (root) {
    if(root.val == null) return 0
    let queue = [{ node: root, max: root.val }]
    let count = 0
    let max = root.val
    while (queue.length > 0) {
        let length = queue.length
        for (let i = 0; i < length; i++) {
            let { node, max } = queue.shift()
            if (node.val >= max) {
                count++
            }
            newmax = Math.max(node.val,max)
            if (node.left != null) {
                queue.push({ node: node.left, max: newmax })
            }
            if (node.right != null) {
                queue.push({ node: node.right, max: newmax })
            }
        }
    }
    return count
};

File: 1544-count-good-nodes-in-binary-tree/1544-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def again(node, max_val):
            if not node:
                return 0

            # Check if the current node is "good"
            count = 1 if node.val >= max_val else 0

            # Update the maximum value for the path
            max_val = max(max_val, node.val)

            # Recursively count "good" nodes in left and right subtrees
            count += again(node.left, max_val)
            count += again(node.right, max_val)
            
            return count

        return again(root, root.val)